src/main/java/org/benetech/servicenet/service/impl/OrganizationMatchServiceImpl.java

```diff
@@ -269,10 +278,19 @@ private void removeMatches(List<OrganizationMatch>  matches) {
         matches.add(save(match));
         matches.add(save(mirrorMatch));
 
+        for (MatchSimilarityDTO similarityDTO : similarityDTOS) {
+            similarityDTO.setOrganizationMatchId(match.getId());
+            matchSimilarityService.save(similarityDTO);
+            similarityDTO.setOrganizationMatchId(mirrorMatch.getId());
+            matchSimilarityService.save(similarityDTO);
```

Comment:
why it's saved twice?

Reply:
For match and mirrorMatch (as we discussed, it is better to keep both versions - same as OrganizationMatch)

Reply:
ok